data/domain/alerts.tsx

```diff
@@ -315,37 +315,50 @@ const getPlayerAlerts = (player: Player, anvil: AnvilWrapper, playerObols: Obol[
-                alerts.push(new DivinityStyleAlert(player, "You should use Mindful style"));
-            } else if ((player.skills.get(SkillsIndex.Divinity)?.level ?? 0) >= 60 && (playerDivinityData.style.name ?? '') != 'Zen') {
-                // If using Mantra on 8 or more players it'll be better than using Zen for EXP
-                if (!(divinityData.playerInfo.filter(info => (info.style.name ?? '') == 'Mantra').length >= 8 && (playerDivinityData.style.name ?? '') == 'Mantra')) {
-                    alerts.push(new DivinityStyleAlert(player, "You should use Zen style"));
-                }
-            } else if ((player.skills.get(SkillsIndex.Divinity)?.level ?? 0) >= 25) {
-                // If using Mantra on 7 or more players it'll be better than using Vitalic for EXP
-                if (!(divinityData.playerInfo.filter(info => (info.style.name ?? '') == 'Mantra').length >= 7 && (playerDivinityData.style.name ?? '') == 'Mantra')) {
-                    alerts.push(new DivinityStyleAlert(player, "You should use Vitalic style"));
-                }
-            } else if ((player.skills.get(SkillsIndex.Divinity)?.level ?? 0) >= 10 && (playerDivinityData.style.name ?? '') == 'Kinesis') {
-                alerts.push(new DivinityStyleAlert(player, "You should use Focus style"));
-            } else if ((player.skills.get(SkillsIndex.Divinity)?.level ?? 0) >= 5 && (playerDivinityData.style.name ?? '') == 'Kinesis') {
-                alerts.push(new DivinityStyleAlert(player, "You should use Chakra style"));
-            }            
-            break;
         case Activity.Unknown:
             alerts.push(new DoingNothingAlert(player));
             break;
     }
 
-    // Separate those two from above to avoid having it in multiple case
+
+    // Separate those three blocks from above to avoid having it in multiple case
     if (player.getActivityType() != Activity.Lab && (playerDivinityData.gods.length ?? 0) == 1 && playerDivinityData.gods[0].data.name == 'Goharut') {
         alerts.push(new DivinityLinkedAlert(player, "Goharut god is useless while you're not in lab"));
     }
-    if (player.getActivityType() != Activity.Divinity && (playerDivinityData.style.name ?? '') != 'TranQi' && (player.skills.get(SkillsIndex.Divinity)?.level ?? 0) >= 40) {
+    if (player.getActivityType() != Activity.Divinity && (playerDivinityData.style.name ?? '') != 'TranQi' && !(player.getActivityType() == Activity.Lab && playerDivinityData.gods.some(god => god.data.name == 'Goharut')) && (player.skills.get(SkillsIndex.Divinity)?.level ?? 0) >= 40) {
         alerts.push(new DivinityStyleAlert(player, "You should use TranQi style while not meditating"));
     }
+    if (player.getActivityType() == Activity.Divinity || (player.getActivityType() == Activity.Lab && playerDivinityData.gods.some(god => god.data.name == 'Goharut'))) {
```

Comment:
I'm using the gods.some whenever it's useful to future-proof in cases where multiple gods are linked and Doot isn't owned, and also checking if Doot is owned instead of gods.length

Reply:
I might be wrong since I haven't looked at this code in a while but I don't think you need to look at doot/companion data.

[This code](https://github.com/Sludging/idleon-efficiency/blob/7c7aeaea28389ce46665b536aedf514dd2b3a4ac/data/domain/companions.tsx#L80-L88) will update the god array to have all gods if doot is owned and is one of the first few things that run so it should be updated prior to alerts code being inspected.

So in case of doot (or ES, or any other god altering feature) you can just look at the god array ... I think 😆